adding data integration (loading data from pretalx with a 1min ttl)

martinheidegger opened the pull request:
This PR adds basic data integration from pretalx to the netlify site generator. It loads all session and user data (including images & resources).

As the schedule is using the https://conf.consento.org/template-test schedule data for the visualization which has some tricks in there that should work.

For the test, the data is still `.gitignore`ed to make sure that the git repository is not spammed with commits - so you need to run `npm run build` in the root directory after `npm install`

ssaavedra committed:
Create gitkeep to keep the content folder when running pretalx-fetch

martinheidegger committed:
adding data integration (loading data from pretalx with a 1h ttl)

martinheidegger committed:
Added resource display

martinheidegger committed:
Using template-test resource for better testing

martinheidegger committed:
Adding ical link

martinheidegger committed:
More markdown support

martinheidegger committed:
Reducing ttl to 60 seconds

ssaavedra committed:
Add root note_modules to gitignore

ssaavedra committed:
Add layout to the new liquid files for pretalx

ssaavedra committed:
Link the layout in the navbar

ssaavedra commented on package.json:
```diff
@@ -0,0 +1,10 @@
+  "private": "true",
+  "scripts": {
+    "build": "netlify build",
+    "postinstall": "cd website; echo 'installing website dependencies'; npm i; echo 'installing plugin dependencies'; cd plugins/pretalx-fetch; npm i"
```
Why not leave the postinstall for the plugin dependency on website/package.json?

ssaavedra commented:
I cannot run `netlify build` or `npm i` on the root folder because I'm not the Netlify account holder :-(

martinheidegger committed:
Extracting pretalx fetch also in a binary to be executed in the website folder

martinheidegger committed:
Better automation of build system

martinheidegger commented:
@ssaavedra I updated the build scripts to run smoother and without netlify (you should be able to just run `npm run dev` in the dev folder) - netlify deploy example: https://deploy-preview-25--dat-events.netlify.app/2020/schedule/

martinheidegger commented:
I looked a bit into calendar visualization. [fullcalendar](https://fullcalendar.io/) is the visualization that pretalx uses but in [tui](https://github.com/nhn/tui.calendar/blob/master/docs/getting-started.md) it should be easily possible to add multiple timezones.

ssaavedra committed:
Add luxon.min.css for future handling of dates

ssaavedra committed:
Update 2020 layout

- Add calendar handling base javsacript
- Link to speakers page in navbar
- Add calendar.js to handle calendar stuff

ssaavedra committed:
Style proposal iteration

This is a draft of the style for the agenda and speakers

The speakers is still not closed and the talks page may be improved
for accessiblity and better readability.

The agenda currently looks the user's timezone.

We want to add a timezone selector to change the timezones everywhere.
The calendar.js file is already prepared for that change when we make it
happen in the style pages.

ssaavedra committed:
Fix heading level for readibility purposes

ssaavedra committed:
Change color in speaker border

Green was too heavy on the eyes.
This page still needs a design revamp, we might try to invert the colors
like in the index/agenda ones.

martinheidegger committed:
Using the real dat event

martinheidegger committed:
lint fix

martinheidegger committed:
Renamed "api" to "paging" as api was really hard to understand :-)

martinheidegger committed:
Added fetching for event data

martinheidegger committed:
Added support for optional additional css files

martinheidegger committed:
Added first iteration on calendar display

martinheidegger committed:
Fixed missing comma

martinheidegger committed:
Updating nvmrc

martinheidegger committed:
Skipping v prefix

martinheidegger committed:
Changing to .node_version

martinheidegger committed:
Needs to be in base directory

martinheidegger committed:
fix

martinheidegger committed:
Adding padding to other page sections (for making sure that the text gets padding on mobile devices

martinheidegger committed:
Using markdown for the header in schedule

martinheidegger committed:
fill-available prevents left-right scrolling on mobile

martinheidegger commented:
I think the schedule display should work as-is, now the speaker/talk/speakers/talks displays need some love.

martinheidegger committed:
Removing padding for schedule page.

martinheidegger committed:
no message

martinheidegger committed:
Removed obsolete agenda

martinheidegger committed:
Renamed speaker -> person and speakers -> people

martinheidegger committed:
Sorting people by name

martinheidegger committed:
Fixing assets url

martinheidegger committed:
typo

martinheidegger committed:
Updating speaker display.

martinheidegger committed:
Removed console.log

Frando commented:
I was finally able to properly test this out - it looks great! Thanks a lot martin for all the work done here. From initial review this looks great to be merged and we can still improve in the process.

Frando commented:
I see there is some code to change the timezone in `calender.js` but it seems it's not yet active - is this an unfinished part or broken? I can look into it today.

martinheidegger committed:
Adding support for image urls that don't have a proper extension.

martinheidegger committed:
Removing speakers without presentations

martinheidegger committed:
Light refactoring.

martinheidegger committed:
Updating style of speaker shapes

martinheidegger committed:
Fixed bleeding

martinheidegger committed:
Updating line-height for speaker-names (for mobile displays)

martinheidegger committed:
Extracted speaker image to filter

martinheidegger committed:
Adding correct speaker image to person display

martinheidegger committed:
Added basic rollup - yeah

martinheidegger committed:
Added pre-post js files to layout

martinheidegger committed:
cleanup

martinheidegger committed:
missing export

ssaavedra committed:
Remove calendar.js and use Temporal instead for talks for the moment

- Use a filter in Eleventy to show times without the dates
- Remove calendar.js and luxon from assets, which are no longer used
- Add timezone to the talks page

ssaavedra committed:
Fix link to speaker bio in talk page

martinheidegger commented on website/content/2020/schedule.11ty.js:
```diff
@@ -55,7 +55,7 @@ function list (iterable, mapper, sep='', lastSep) {
 }
 
-module.exports = class Schedule {
+class Schedule {
```
Does this work?

ssaavedra replied:
No, it shouldn't have cleanly merged. This was a mistake.

ssaavedra committed:
Reinstate back module.exports

ssaavedra committed:
Style the speakers in talk/:id and person/:id

Now the speakerImage filter does not include the name anymore
This allows for more flexible naming placement

The speaker list in the talks now include the avatar

The resources have been unlisted from the public view

ssaavedra committed:
Change idiom of speaker -> conductor

ssaavedra committed:
Work around the speakers.avatar problem for gravatars

ssaavedra committed:
Add a margin to paragarphs in the schedule page

ssaavedra committed:
Center the heading with each day in the schedule

dpaez commented:
Looks great! <3
